src/Controller/Component/CommonComponent.php

```diff
@@ -343,4 +344,22 @@ public function isForeignReferer($ref = null) {
 		return true;
 	}
 
+	/**
+	 * Similar to allowMethod(), but allows only certain extensions.
+	 * Use '' to allow only non-ext.
+	 *
+	 * @param array<string>|string $extensions
+	 * @return void
+	 */
+	public function allowExtension(array|string $extensions): void {
```

Comment:
I'm aware you adapted from `allowMethod`, which is singular. But I can only assume this will often be more than one extension.

Thus, I would rename it to `allowExtensions`.